packages/app/lib/ui/widgets/note_field.dart

```diff
@@ -0,0 +1,142 @@
+
+  @override
+  Widget build(BuildContext context) {
+    final t = AppLocalizations.of(context)!;
+
+    final initialValue = widget.value ?? '';
+
+    return Form(
+        key: _formKey,
+        child: MeliCard(
+            child: Column(children: [
+          MeliCardHeader(
+              title: t.noteCardTitle,
+              icon: _inputMode == InputMode.read
+                  ? CardActionButton(
+                      icon: Icon(Icons.edit),
+                      onPressed: _handleStartEditMode,
+                    )
+                  : null),
+          Container(
+              padding: EdgeInsets.symmetric(vertical: 10.0, horizontal: 6.0),
+              alignment: AlignmentDirectional.centerStart,
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  // TODO: TextField seems to have a bug where it does not respect _textInputFocusNode.canRequestFocus being false when in 'read' input mode
+                  TextFormField(
+                      controller: _textInputController,
+                      decoration: InputDecoration(border: OutlineInputBorder()),
+                      focusNode: _textInputFocusNode,
+                      keyboardType: TextInputType.multiline,
+                      maxLines: null,
+                      minLines: 5,
+                      onSaved: (text) => _handleSubmit(text, initialValue),
+                      readOnly: _inputMode == InputMode.read,
+                      textCapitalization: TextCapitalization.sentences),
+                  if (_inputMode == InputMode.edit) ...[
+                    SizedBox(height: 12),
+                    Row(children: [
+                      OverflowBar(
+                        spacing: 12,
+                        overflowAlignment: OverflowBarAlignment.start,
+                        children: [
+                          FilledButton(
+                              onPressed: _handleSave,
+                              child: Text(t.noteCardSaveAction)),
+                          OutlinedButton(
+                              onPressed: _handleCancel,
+                              child: Text(t.noteCardCancelAction))
+                        ],
+                      )
+                    ])
+                  ]
+                ],
+              )),
+        ])));
+  }
+
+  void _handleCancel() {
+    setState(() {
+      _formKey.currentState!.reset();
+      _inputMode = InputMode.read;
+      _textInputFocusNode.canRequestFocus = false;
+    });
+  }
+
+  void _handleSubmit(String? text, String initialValue) {
+    final trimmed = (text ?? '').trim();
+
+    if (initialValue == trimmed) {
+      _handleCancel();
+      return;
+    }
+
+    setState(() {
+      _inputMode = InputMode.read;
+      _textInputFocusNode.canRequestFocus = false;
+      widget.onUpdate.call(trimmed);
+    });
+  }
+
+  void _handleSave() {
+    _formKey.currentState!.save();
+  }
```

Comment:
kind of a verbose approach when it comes to working with a single form element but allows us to rely less on reading the controller value directly. maybe not a worthy tradeoff since we need the controller anyways, so open to simplifying a bit